xray/utils/analyzermanager.go

```diff
@@ -76,10 +95,20 @@ func (am *AnalyzerManager) ExistLocally() (bool, error) {
 	return fileutils.IsFileExists(analyzerManagerPath, false)
 }
 
-func (am *AnalyzerManager) Exec(configFile string) error {
-	cmd := exec.Command(am.analyzerManagerFullPath, applicabilityScanCommand, configFile)
+func (am *AnalyzerManager) Exec(configFile string, scanCommand string) error {
+	var err error
+	cmd := exec.Command(am.analyzerManagerFullPath, scanCommand, configFile)
+	defer func() {
+		if !cmd.ProcessState.Exited() {
+			if killProcessError := cmd.Process.Kill(); errorutils.CheckError(killProcessError) != nil {
+				log.Info(fmt.Sprintf("failed to kill analyzer manager process: %s", killProcessError.Error()))
+				err = errors.Join(err, killProcessError)
```

Comment:
This line requires using a named returned value for the error returned by the function.